diff -urNp Mesa-6.4.2/src/mesa/drivers/glide/fxddspan.c Mesa-6.4.2y/src/mesa/drivers/glide/fxddspan.c
--- Mesa-6.4.2/src/mesa/drivers/glide/fxddspan.c	2005-05-10 01:14:58.000000000 +0300
+++ Mesa-6.4.2y/src/mesa/drivers/glide/fxddspan.c	2015-08-27 20:30:08.000000000 +0300
@@ -580,6 +580,7 @@ fxSetupDDSpanPointers(GLcontext * ctx)
 
    swdd->SetBuffer = fxDDSetBuffer;
 
+#if 0 /* these members removed in 6.4 */
    switch (fxMesa->colDepth) {
           case 15:
                swdd->WriteRGBASpan = tdfxWriteRGBASpan_ARGB1555;
@@ -631,6 +632,7 @@ fxSetupDDSpanPointers(GLcontext * ctx)
       swdd->WriteStencilPixels = fxWriteStencilPixels;
       swdd->ReadStencilPixels = fxReadStencilPixels;
    }
+#endif
 #if 0
    swdd->WriteCI8Span		= NULL;
    swdd->WriteCI32Span		= NULL;
diff -urNp Mesa-6.4.2/src/mesa/x86/assyntax.h Mesa-6.4.2y/src/mesa/x86/assyntax.h
--- Mesa-6.4.2/src/mesa/x86/assyntax.h	2005-11-04 02:44:32.000000000 +0200
+++ Mesa-6.4.2y/src/mesa/x86/assyntax.h	2015-08-27 20:27:51.000000000 +0300
@@ -1736,7 +1736,7 @@ SECTION _DATA public align=16 class=DATA
  */
 #if defined(GNU_ASSEMBLER) && !defined(__DJGPP__) && !defined(__MINGW32__)
 #  define HIDDEN(x) .hidden x
-#elif defined(__GNUC__) && (__GNUC__ * 100 + __GNUC_MINOR__) >= 303 && !defined(__MINGW32__)
+#elif defined(__GNUC__) && (__GNUC__ * 100 + __GNUC_MINOR__) >= 303 && !defined(__MINGW32__) && !defined(__DJGPP__)
 #  pragma GCC visibility push(default)
 #  define HIDDEN(x) .hidden x
 #else
diff -urNp Mesa-6.4.2/Makefile.DJ Mesa-6.4.2y/Makefile.DJ
--- Mesa-6.4.2/Makefile.DJ	2005-05-24 01:02:53.000000000 +0300
+++ Mesa-6.4.2y/Makefile.DJ	2015-08-27 20:35:34.000000000 +0300
@@ -59,18 +59,14 @@
 
 GLU ?= mesa
 
-CFLAGS = -Wall -W -pedantic
+CFLAGS = -Wall
 CFLAGS += -O2 -ffast-math
 
 export CFLAGS
 
-ifeq ($(wildcard $(addsuffix /rm.exe,$(subst ;, ,$(PATH)))),)
-UNLINK = del $(subst /,\,$(1))
-else
-UNLINK = $(RM) $(1)
-endif
+UNLINK = rm -f $(1)
 
-all: libgl libglu libglut
+all: libgl libglu
 
 libgl: lib
 	$(MAKE) -f Makefile.DJ -C src/mesa
@@ -84,9 +80,9 @@ lib:
 
 clean:
 	$(MAKE) -f Makefile.DJ clean -C src/mesa
-	$(MAKE) -f Makefile.DJ clean -C src/glu/mesa
-	$(MAKE) -f Makefile.DJ clean -C src/glu/sgi
-	$(MAKE) -f Makefile.DJ clean -C src/glut/dos
+	-$(MAKE) -f Makefile.DJ clean -C src/glu/mesa
+	-$(MAKE) -f Makefile.DJ clean -C src/glu/sgi
+	-$(MAKE) -f Makefile.DJ clean -C src/glut/dos
 
 realclean: clean
 	-$(call UNLINK,lib/*.a)
diff -urNp Mesa-6.4.2/src/glu/sgi/Makefile.DJ Mesa-6.4.2y/src/glu/sgi/Makefile.DJ
--- Mesa-6.4.2/src/glu/sgi/Makefile.DJ	2004-12-15 10:00:44.000000000 +0200
+++ Mesa-6.4.2y/src/glu/sgi/Makefile.DJ	2015-08-27 20:34:10.000000000 +0300
@@ -58,13 +58,7 @@ CXXFLAGS = $(CFLAGS) -Ilibnurbs/internal
 AR = ar
 ARFLAGS = crus
 
-HAVEDXE3 = $(wildcard $(DJDIR)/bin/dxe3gen.exe)
-
-ifeq ($(wildcard $(addsuffix /rm.exe,$(subst ;, ,$(PATH)))),)
-UNLINK = del $(subst /,\,$(1))
-else
-UNLINK = $(RM) $(1)
-endif
+UNLINK = rm -f $(1)
 
 C_SOURCES = \
 	libutil/error.c \
@@ -174,11 +168,7 @@ $(LIBDIR)/$(GLU_LIB): $(OBJECTS)
 	$(AR) $(ARFLAGS) $@ $^
 
 $(LIBDIR)/$(GLU_DXE) $(LIBDIR)/$(GLU_IMP): $(OBJECTS)
-ifeq ($(HAVEDXE3),)
-	$(warning Missing DXE3 package... Skipping $(GLU_DXE))
-else
 	-dxe3gen -o $(LIBDIR)/$(GLU_DXE) -Y $(LIBDIR)/$(GLU_IMP) -D "MesaGLU/SGI DJGPP" -E _glu -P gl.dxe -U $^
-endif
 
 clean:
 	-$(call UNLINK,libutil/*.o)
diff -urNp Mesa-6.4.2/src/mesa/Makefile.DJ Mesa-6.4.2y/src/mesa/Makefile.DJ
--- Mesa-6.4.2/src/mesa/Makefile.DJ	2004-12-15 10:00:44.000000000 +0200
+++ Mesa-6.4.2y/src/mesa/Makefile.DJ	2015-08-27 20:34:36.000000000 +0300
@@ -63,6 +63,8 @@ GL_IMP = libigl.a
 export LD_LIBRARY_PATH := $(LD_LIBRARY_PATH);$(LIBDIR);$(GLIDE)/lib
 
 CC = gcc
+# HOST_CC is for generating X86/gen_matypes.exe when cross-compiling.
+HOST_CC = gcc
 CFLAGS += $(INCLUDE_DIRS)
 CFLAGS += -DUSE_EXTERNAL_DXTN_LIB=1
 ifeq ($(FX),1)
@@ -76,13 +78,7 @@ endif
 AR = ar
 ARFLAGS = crus
 
-HAVEDXE3 = $(wildcard $(DJDIR)/bin/dxe3gen.exe)
-
-ifeq ($(wildcard $(addsuffix /rm.exe,$(subst ;, ,$(PATH)))),)
-UNLINK = del $(subst /,\,$(1))
-else
-UNLINK = $(RM) $(1)
-endif
+UNLINK = rm -f $(1)
 
 include sources
 
@@ -133,15 +129,11 @@ $(LIBDIR)/$(GL_LIB): $(OBJECTS)
 	$(AR) $(ARFLAGS) $@ $^
 
 $(LIBDIR)/$(GL_DXE) $(LIBDIR)/$(GL_IMP): $(OBJECTS)
-ifeq ($(HAVEDXE3),)
-	$(warning Missing DXE3 package... Skipping $(GL_DXE))
-else
 ifeq ($(FX),1)
 	-dxe3gen -o $(LIBDIR)/$(GL_DXE) -Y $(LIBDIR)/$(GL_IMP) -D $(LIBNAME) -E _gl -E _DMesa -P glide3x.dxe -U $^
 else
 	-dxe3gen -o $(LIBDIR)/$(GL_DXE) -Y $(LIBDIR)/$(GL_IMP) -D $(LIBNAME) -E _gl -E _DMesa -U $^
 endif
-endif
 
 $(X86_OBJECTS): x86/matypes.h
 
@@ -149,7 +141,7 @@ x86/matypes.h: x86/gen_matypes.exe
 	$< > $@
 
 x86/gen_matypes.exe: x86/gen_matypes.c
-	$(CC) -o $@ $(CFLAGS) -s $<
+	$(HOST_CC) -o $@ $(CFLAGS) -s $<
 
 clean:
 	-$(call UNLINK,array_cache/*.o)
